### PR TITLE
feat: add unit tests for empty widgets

### PR DESCRIPTION
**Pull request by KleinMichalGit**
## Frontend Pull Request Description

add unit tests for empty widgets

## Checklist

- [x] My code follows the [style guidelines](https://github.com/solarwinds/nova/blob/main/docs/STYLE_GUIDE.md) of this project
- [x] I have performed a self-review of my code
- [ ] I have updated [change log](https://github.com/solarwinds/nova/blob/main/docs/CHANGELOG.md)
- [x] I have been following [Definition of done](https://github.com/solarwinds/nova/blob/main/docs/DEFINITION_OF_DONE.md)
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new lint warnings
- [x] New and existing unit tests pass locally and on CI with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/dashboards/src/lib/components/proportional-widget/proportional-widget.component.ts [hunk 1]
```diff
@@ -496,4 +496,12 @@ export class ProportionalWidgetComponent
 
         return colorProvider;
     }
+
+    public get isEmpty(): boolean {
+        return (
+            !this.widgetData ||
+            this.widgetData.length === 0 ||
```
Comment (pilat-martin) [inline packages/dashboards/src/lib/components/proportional-widget/proportional-widget.component.ts:503]: What kind of data does proportional widget take? Is it safe to call `.length`?

**Comment by erzik987** [comment]
This is from version 15.0.9
image

Its not breaking change but it looks like the picture is not scaling very well. Can you try to address this ? If not create a bug

File: packages/dashboards/src/lib/components/proportional-widget/proportional-widget.component.html [hunk 2]
```diff
@@ -7,10 +7,7 @@
         'empty-grid': !(widgetData && widgetData.length > 0 && chartAssist)
     }"
 >
-    <div
-        class="d-flex chart has-overlay"
-        *ngIf="widgetData && widgetData.length > 0 && chartAssist"
-    >
+    <div class="d-flex chart has-overlay" *ngIf="!isEmpty">
```
[the remark this answers is not in the document]
Comment (KleinMichalGit) [reply packages/dashboards/src/lib/components/proportional-widget/proportional-widget.component.html:10]: there is no removed code. Duplicate condition was moved into component. nothing changed

**Comment by KleinMichalGit** [comment]
this is from oolatest (before propagation 15.0.9 nova into fusion) and empty states do not adjust vertically at all for any widget. Will contact UX
image

File: packages/dashboards/src/lib/components/list-widget/list-widget.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import { ComponentFixture, TestBed, waitForAsync } from "@angular/core/testing";
+
+import { EventBus, IEvent } from "@nova-ui/bits";
+import {
+    ListWidgetComponent,
+    PIZZAGNA_EVENT_BUS,
+    ProviderRegistryService,
+} from "@nova-ui/dashboards";
+
+describe(ListWidgetComponent.name, () => {
```
[the remark this answers is not in the document]
Comment (KleinMichalGit) [reply packages/dashboards/src/lib/components/list-widget/list-widget.component.spec.ts:10]: I added some use cases

**Comment by KleinMichalGit** [comment]
new tickets for Drilldown (list) widget behavior when passed array of booleans and Vertical scaling of empty widgets were created. The scope of this PR should be adding unit tests only.

**Comment by KleinMichalGit** [comment]
https://swicloud.atlassian.net/browse/NUI-6240
https://swicloud.atlassian.net/browse/NUI-6241

**Review by pilat-martin — APPROVED** [review]
